server.js

```diff
@@ -20,11 +20,67 @@ const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+
+// Start defining your routes here
+//endpoint to output movies by title, release year, cast and country
+app.get('/movies', (req, res) => {
+  const { title, year, cast, country} = req.query
+  let moviesToSend = netflixData
+
+  if (title) {
+    moviesToSend = moviesToSend
+      .filter(movie => movie.title.toString().toLowerCase().includes(title.toLowerCase()))
+  } 
+
+  if (year) {
+    moviesToSend = moviesToSend
+      .filter(movie => movie.release_year === +year)
+  }
+
+  if (cast) {
+    moviesToSend = moviesToSend
+      .filter(movie => movie.cast.toString().toLowerCase().includes(cast.toLowerCase()))
+  }
+
+  if (country) {
+    moviesToSend = moviesToSend
+      .filter(movie => movie.country.toString().toLowerCase() === country)
+  }
+
+  res.json(moviesToSend)
+})
+
+//endpoint to output possible types of movies
+app.get('/movies/types', (req, res) => {
+  const typesDuplicated = netflixData.map(item => item.type)
+
+  const uniqueTypes = []
+  typesDuplicated.forEach(item => {
+    if (!uniqueTypes.includes(item)) {
+      uniqueTypes.push(item)
+    }
+  })
+  res.json(uniqueTypes)
+})
+
+//endpoint to get one specific movie by id
+app.get('/movies/id/:id', (req, res) => {
```

Comment:
it is a convention that when you are using the primal key (id here) you need to use route: `/movies/:id`